applications/serial_lte_modem/doc/slm_extending.rst

```diff
@@ -0,0 +1,50 @@
+.. _slm_extending:
+
+Extending the application
+#########################
+
+You can extend the serial LTE modem application by adding your own AT commands and custom error handling.
+
+Adding AT commands
+******************
+
+If you want to implement custom AT commands and add them to the serial LTE modem application, see the parser implementation of the provided proprietary AT commands for reference.
+The source files can be found in the :file:`applications/nrf9160/serial_lte_modem/src/` folder.
+
+Complete the following steps to add a parser for your own AT commands:
+
+1. Create a header file in :file:`applications/nrf9160/serial_lte_modem/src/`.
+   The file must expose the following functions:
+
+   * ``*_init()`` - Initialize the parser.
+   * ``*_uninit()`` - Uninitialize the parser.
+   * ``*_parse()`` - Parse an AT command string.
+   * ``*_clac()`` - List all defined AT commands.
+
+   See the files for existing AT command parsers for reference.
+#. Implement your AT strings and handlers in a corresponding :file:`.c` file.
+   See the files for existing AT command parsers for reference.
+
+   Pay attention to the following requirements:
+
+   * The names of new AT commands should start with ``AT#X``.
+   * Before entering idle state, the serial LTE modem application will call the uninit function.
+     When exiting idle state, it will call the init function.
+     Make sure that the uninit function exits successfully.
```

Comment:
Is this correct? I'm not sure I understood correctly why users need to pay special attention to the uninit function.

Reply:
If error occurs in uninit during enter idle state, SLM will fail to enter idle state.

@junqingzou please confirm.

Reply:
Yes, SLM needs to shutdown its own functions before entering IDLE or SLEEP. Another recent PR will reboot to apply FOTA downloaded image, which also requires graceful shutdown of SLM functions.